File: Dante2/Module2.04/Zadanie4.03/main.c
#include <stdio.h>
#include <stdlib.h>

float* create_array(int N)
{
    if(N <= 0)
    {
        return NULL;
    }

    float *array = malloc(sizeof(float) * N);
    if(!array)
    {
        return NULL;
    }

    return array;
}

int main()
{
    int num;
    printf("Enter: ");
    int check = scanf("%i",&num);
    if(check != 1)
    {
        printf("Incorrect input");
        return 1;
    }
    if(num <= 0)
    {
        printf("Incorrect input data");
        return 2;
    }

    float *array = create_array(num);
    if(array == NULL)
    {
        printf("Failed to allocate memory");
        return 8;
    }

    printf("Enter: ");
    for(int i = 0; i < num; i++)
    {
        int check1 = scanf("%f",&*(array + i));
        if(check1 != 1)
        {
            printf("Incorrect input");
            free(array);
            return 1;
        }
    }

    for(int i = num - 1; i >= 0; i--)
    {
        printf("%f ",*(array + i));
    }

    free(array);
    return 0;
}

File: Dante2/Module2.04/Zadanie4.03/main.rs
fn create_array(size: i32) -> Vec<i32> {
    if size <= 0 {
        panic!("Incorrect input");
    }

    vec![0; size as usize]
}
fn main() {
    let mut input: String = String::new();

    println!("Enter: ");
    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let number: i32 = input.trim().parse().expect("Incorrect parse");

    if number <= 0 {
        panic!("Incorrect input data");
    }

    let mut array: Vec<i32> = create_array(number);

    print!("Enter: ");
    for i in 0..array.len()
    {
        input.clear();
        std::io::stdin().read_line(&mut input).expect("Incorrect input");
        let num: i32 = input.trim().parse().expect("Incorrect parse");
        array[i] = num;
    }

    for i in (0..number).rev() {
        print!("{} ", array[i as usize]);
    }
}
